mmcv/runner/dist_utils.py

```diff
@@ -68,8 +67,7 @@ def _init_dist_pytorch(backend: str, **kwargs) -> None:
             world_size=int(os.environ['WORLD_SIZE']),
             **kwargs)
     else:
-        num_gpus = torch.cuda.device_count()
-        torch.cuda.set_device(rank % num_gpus)
+        torch.cuda.set_device(os.environ['LOCAL_RANK'])
```

Comment:
I guess this environment variable is only set when using `torchrun`. For some cases like `torch.distributed.launch` this might be absent. So maybe `rand % num_gpus` can be a fallback solution here?

Besides, slurm may have the same issue. I've found a [documentation](https://hpcc.umd.edu/hpcc/help/slurmenv.html) that should be relevant, but haven't tested. Could you solve it in the same issue?

Reply:
Thanks for your advice. I think you give a better solution.
This is my first commit for mmcv, but I don't know why I can't pass the license/cla check, I have signed CLA.

Reply:
I ported your code form mmengine to mmcv, but I have few knowledge about slurm, could help to review this snippet？ Thank you.

Reply:
> I'm afraid this environment variable is only set when using `torchrun`. For some cases like `torch.distributed.launch` this might be absent.

I experimented with PyTorch 1.6 (which is the minimum supported version in OpenMMLab repos). It actually **has 
'LOCAL_RANK' environment variable**. What it doesn't have is 'LOCAL_WORLD_SIZE'. So I'm wrong and I guess we don't need the fallback here.